deploy/deploy.js

```diff
@@ -0,0 +1,144 @@
+
+        console.log(`Deployment finished successfully ${new Date().toUTCString()}`);
+    }
+    catch (ex) {
+        console.log(ex);
+        if (ex.response) {
+            console.log(ex.response.data.error.message);
+        }
+    }
+})();
+
+async function deployStorageQueues(storageSessionToken) {
+
+    console.log("Deploying azures storage queues");
+
+    let storageHttpConfig = {
+        headers: {
+            "x-ms-version": "2018-03-28",
+            "x-ms-date": new Date().toUTCString()
+        }
+    }
+
+    for (let queue of require("./definitions/storage/queues.config.json")) {
+        let timeout = queue.timeout ? `&timeout=${queue.timeout}` : "";
+        await put(`https://${config.storageAccountName}.queue.core.windows.net/${queue.name}${timeout}`, null, storageSessionToken, storageHttpConfig);
+    }
+}
+
+async function deployAzureFunctions(armSessionToken) {
+
+    console.log("Deploying azure functions");
+    
+    // deploy app settings to the deployment slot
+    let cosmosKeyResponse = await post(azureResourceUrl("Microsoft.DocumentDB/databaseAccounts", `${config.cosmosAccountName}/listKeys`, "2015-04-08"), null, armSessionToken);
+    await deployArmTemplate("appsettings", armSessionToken, { cosmosMasterKey: cosmosKeyResponse.data.primaryMasterKey });
+
+    // get publish profile credentials
+    let publishProfilesResult = await post(azureResourceUrl("Microsoft.Web/sites", `${config.functionAppName}/slots/swap/config/publishingcredentials/list`, "2016-08-01"), null, armSessionToken);
+
+    let { publishingUserName, publishingPassword } = publishProfilesResult.data.properties;
+
+    var zip = new AdmZip();
+    zip.addLocalFolder(`${__dirname}/../api/dist`);
+
+    // zip deployment of api resources to the deployment slot
+    await post(`https://${config.functionAppName}-swap.scm.azurewebsites.net/api/zipdeploy`, zip.toBuffer(), null, {
+        auth: {
+            username: publishingUserName,
+            password: publishingPassword
+        }
+    })
+
+    //TODO activate setting for disabled functions, possibly reset the instances
+
+    // swap the slots, make the deployment slot the new live slot
+    let swapResponse = await post(azureResourceUrl("Microsoft.Web/sites", `${config.functionAppName}/slotsswap`, "2016-08-01"), { "targetSlot": "swap" }, armSessionToken)
+
+    // wait for swap to complete
+    while (true) {
+        let verifySwapResponse = await get(swapResponse.headers.location, armSessionToken);
+        if (verifySwapResponse.status === 202) {
+            // still running, wait a bit and try again
+            await asyncTimeout(5000);
+            continue;
+        }
+        if (verifySwapResponse.status === 200) {
+            // done
+            break;
+        }
+        // unsupported scenario
+        throw new Error(`Unexpected response ${verifySwapResponse.status} returned while verifying swap of server slots`);
+    }
+
+    //TODO disable the deployment slot
```

Comment:
Why do we want to disable the Staging/Deployment slot? In case there is an immediate need to rollback, this would increase the time to rollback.

Reply:
As long as using sticky slot settings to manage the disabled state doesn't require warm up time after the swap has occurred, we should use that. Will experiment with both.